### DISCUSSION TITLE
Where can I see a list of inputs and outputs in the GUI?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#3740] [General]
Silly question: I can find the input and outputs listed under ITerms for a macro, but where can I find the list of inputs and outputs for the top level design?

Is there somewhere I can find ITerms for the top level design in the GUI?

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/10437d70-44fd-4d47-8e7b-ade917810eab)

**Comment by maliberty** [comment]
On the block:
![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/761514/3c5f0606-31ec-4be6-ae11-aa3212281780)

You can follow the link on any net or instance to inspect the block.

**Reply by oharboe** [reply]
Gotit.

1. click anywhere
2. This will display what is in the Inspector in the image below
3. Click MockArray (the block)
4. Now you have BTerms.

Silly question... What are ITerms and what are BTerms? Instance terms and bus terminals? Why does MockArray only have BTerms and why does the Element only have ITerms?


![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/fb82e2dd-0bda-4694-8b72-20d8794c62bf)

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/f15aaa96-882b-49e4-89db-4d354cba6deb)

**Reply by Iyury1** [reply]
There is a simple text diagram in comments of db.h in a few places that outlines the difference between BTerms and ITerms (line 1796, line 3130, etc.). Block terminals are for external connections, whereas Instance terminals are for connections between internal blocks within a block hierarchy.

I havent looked closely at MockArray and Element yet so I cant answer the rest of your question.

**Reply by oharboe** [reply]
Thanks!  Including link to docs here:

https://github.com/The-OpenROAD-Project/OpenROAD/blob/0e07b05a950cc8eb5c2c088d6e10484f2aa8b1a5/src/odb/include/odb/db.h#L3131
